src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java

```diff
@@ -789,15 +791,19 @@ private NpgResponseException exceptionToNpgResponseException(
 
         if (err instanceof WebClientResponseException e) {
             try {
-                List<ErrorsInnerDto> responseErrors = switch (e.getStatusCode()) {
+
+                HttpStatus httpStatus = HttpStatus.resolve(e.getStatusCode().value());
+                List<ErrorsInnerDto> responseErrors = switch (httpStatus) {
```

Comment:
WebClientResponseException.getStatusCode return an HttpStatusCode implementation.
HttpStatus implements the same interface
So there is no need to perform this resolve against code value operation.
```suggestion

                switch (e.getStatusCode()) {
```

Reply:
sorry, the suggestion was not complete :D, updated it.
Please remove this redundant resolve operation